force-app/main/default/classes/SOQL.cls

```diff
@@ -270,12 +369,21 @@ public with sharing class SOQL {
      * @return List of contacts with the same Mailing State as the Account Billing State
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
-        // Query the Account
+        List<Account> accounts = [
+            SELECT Id, BillingState 
+            FROM Account 
+            WHERE Id = :accountId
+            ];
+
+        String accountBillingState = accounts[0].BillingState;
 
-        // Store the Account's BillingState
+        List<Contact> contacts = [
+            SELECT Id, MailingState 
+            FROM Contact 
+            WHERE MailingState = :accountBillingState
+            ];
```

Comment:
You can also do this as one query, see if you can figure out that syntax. HINT: you can use a query in the criteria